Authors/Igor-Lima/02-graphql-intro/readme.md

```diff
@@ -0,0 +1,379 @@
+// server.js
+var graphql = require ('graphql').graphql
+var express = require('express')
+var graphQLHTTP = require('express-graphql')
+var Schema = require('./schema')
+
+var app = express()
+  .use('/', graphQLHTTP({ schema: Schema, pretty: true }))
+  .listen(8080, function (err) {
+    console.log('GraphQL Server is now running on localhost:8080');
+  });
+```
+
+At this point we have provided two things:
+
+1. a type schema `schema.js`
+2. a server `server.js` to serve queries against a type schema
+
+To run the code, execute `npm install graphql express express-graphql` to get all dependencies and then do `node server.js` to run the program.
+
+```sh
+# install all dependencies
+npm install graphql express express-graphql
+
+# to run the server
+node server.js
+```
+
+When it runs we should have a server up and running in the port `8080`. We can check it out opening the url `http://localhost:8080` in a browser or typing in the command line:
+
+```sh
+$ curl -XPOST -H "Content-Type:application/graphql"  -d 'query { products { id, title, description } }' http://localhost:8080
+{
+  "data": {
+    "products": [
+      {
+        "id": 2354264,
+        "title": "NaVorro Bowman Team Irvin Nike 2016 Pro Bowl Game Jersey - Black",
+        "description": "Get the look of a tried and true football fan with this 2016 Pro Bowl Game jersey and let everyone know you're a die-hard NaVorro Bowman fan! Rock this Nike jersey to watch your favorite player take on the competition. You'll love paying homage to Team Irvin with this jersey."
+      },
+      {
+        "id": 2326907,
+        "title": "Super Bowl 50 Nike Limited Jersey - Black",
+        "description": "You've been watching the Super Bowl for as long as you can remember, and now we are at a milestone: 50 Super Bowls. You can celebrate the golden anniversary of the NFL's biggest game when you grab this Super Bowl 50 limited jersey from Nike. Featuring stunning colors and graphics, you'll be able to show your excitement for the big game everywhere you go when you wear this jersey."
+      }
+    ]
+  }
+}
+```
+
+It's basically how to get GraphQL up and running on the server-side. As we can see, GraphQL is a contract that has been designed with a flexible syntax making the building of client applications easier. Also, we can see how GraphQL challenges the ReST API architecture. For instance, as we know, [a REST API typically delivers all the data a client UI might need about a resource and leaves it up to the client to extract the bits of data it actually wants to show](https://www.compose.io/articles/using-graphql-with-mongodb/). If some of the data needed by the UI isn't provided by a given resource endpoint, then the client needs to go off to the server and request more data from another URL. Over time, the link between front-end applications and back-end servers can become very rigid. As data gets more complex in structure, it gets harder and slower to query with the traditional REST API.
```

Comment:
@igorlima I would suggest removing this paragraph from this article.  This article is a tutorial/introduction for implementing a GraphQL server, not an opinion piece.  The assertions made here in regards to comparisons between ReSTful communication architecture vs GraphQL, doesn't really fit into the topic of the article.  Furthermore some of the assertions made here aren't necessarily true.